en/contributing/plugins.md

```diff
@@ -0,0 +1,102 @@
+# Writing Plugins
+
+DroneCore is split into a [core](https://github.com/dronecore/DroneCore/tree/master/core) and [plugins](https://github.com/dronecore/DroneCore/tree/master/plugins).
+
+Plugins that are located in the correct location (a subfolder of **/plugins**) and have the correct structure are included at compile time. The *cmake* script [autogenerate_plugin_container.cmake](https://github.com/dronecore/DroneCore/blob/master/autogenerate_plugin_container.cmake) takes care of including the plugin folders and integration tests.
+
+> **Note** Plugins can also be defined as part of an SDK/Extension Library (see [SDK Creation](../guide/sdk.md)).
+
+## Plugin Architecture
+
+Plugins should be written so that they are independent of each other (allowing them to be removed/replaced as needed).
+They will still need to be dependent on the core source, and there will be some duplicated functionality across the plugin modules.
+
+The code for each plugin and its unit test (if defined) is stored in a sub-folder of the **plugins** directory. Integration tests for all plugins are stored in **integration_tests**. 
+A simplified view of the folder structure is shown below: 
+
+```
+├── DroneCore
+│   ├── core
+│   ├── integration_tests
+│   └── plugins
+│       ├── action
+│       ├── ...
+│       └── telemetry
+```
+Drilling down, each plugin should have the same structure as shown for the "example" plugin below.
+```
+└── plugins
+    └── example 
+        ├── CMakeLists.txt
+        ├── example.cpp
+        ├── example.h
+        ├── example_impl.cpp
+        ├── example_impl.h
+        └── example_impl_test.cpp  ##optional
+```
+
+## Create a Plugin
+
+To create a new C++ plugin, copy the [example](https://github.com/dronecore/DroneCore/tree/master/external_example/plugins/example/) plugin from the "external_example" into the DroneCore **plugins** directory.

```

Comment:
@julianoes Are they better off copying this external_example for core plugins or one of the existing plugins (e.g. "action"). Why?
The CMakeLists.txt look similar but not identical, which makes me think that copying an existing standard plugin would be better?

Reply:
Hm. The only difference is about external additional libraries. I would say copying from an existing standard plugin makes sense.